bson/bsoncodec/default_value_encoders.go

```diff
@@ -289,29 +289,44 @@ func (dve DefaultValueEncoders) MapEncodeValue(ec EncodeContext, vw bsonrw.Value
 // struct codec.
 func (dve DefaultValueEncoders) mapEncodeValue(ec EncodeContext, dw bsonrw.DocumentWriter, val reflect.Value, collisionFn func(string) bool) error {
 
-	encoder, err := ec.LookupEncoder(val.Type().Elem())
-	if err != nil {
-		return err
+	encoder, encErr := ec.LookupEncoder(val.Type().Elem())
+	if encErr != nil && val.Type().Elem().Kind() != reflect.Interface {
+		return encErr
 	}
 
 	keys := val.MapKeys()
+	var err error
 	for _, key := range keys {
 		if collisionFn != nil && collisionFn(key.String()) {
 			return fmt.Errorf("Key %s of inlined map conflicts with a struct field name", key)
 		}
+
+		currVal := val.MapIndex(key)
+		currEncoder := encoder
+		if currEncoder == nil && (currVal.Kind() == reflect.Interface) {
+			currVal = currVal.Elem()
+			if !currVal.IsValid() {
+				return encErr
+			}
+			currEncoder, err = ec.LookupEncoder(currVal.Type())
```

Comment:
nit: I think it makes more sense to reassign `encErr` here and return that in the error conditional below.

Reply:
added a new error for invalid interface element